modules/components/src/AdvancedFacetView/LiveAdvancedFacetView.js

```diff
@@ -0,0 +1,141 @@
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      ES_HOST: ES_HOST,
+    },
+    body: JSON.stringify({
+      query: query,
+    }),
+  })
+    .then(res => res.json())
+    .then(data => data.data);
+
+const fetchMapping = async fetchConfig =>
+  fetchGraphqlQuery({
+    query: `{
+      ${fetchConfig.ES_INDEX} {
+        mapping,
+      }
+    }`,
+    ...fetchConfig,
+  }).then(data => {
+    return Promise.resolve(data[fetchConfig.ES_INDEX]);
+  });
+
+const fetchExtendedMapping = async fetchConfig =>
+  fetchGraphqlQuery({
+    query: `{
+      ${fetchConfig.ES_INDEX} {
+        extended,
+      }
+    }`,
+    ...fetchConfig,
+  }).then(data => {
+    return Promise.resolve(data[fetchConfig.ES_INDEX]);
+  });
+
+const fetchAggregationDataFromExtendedMapping = async ({
+  extended,
+  PROJECT_ID,
+  ES_INDEX,
+  API_HOST,
+  ES_HOST,
+}) => {
+  const fetchConfig = { PROJECT_ID, ES_INDEX, API_HOST, ES_HOST };
+  const serializeToGraphQl = aggName => aggName.split('.').join('__');
+  const serializeToPath = aggName => aggName.split('__').join('.');
+  const allAggsNames = extended
+    .map(entry => entry.field)
+    .map(serializeToGraphQl);
+  const getAggregationQuery = () =>
+    allAggsNames
+      .map(aggName => {
+        const aggType = extended.find(
+          entry => serializeToGraphQl(entry.field) === aggName,
+        ).type;
+        return `
+          ${aggName} {
+            ${
+              esToAggTypeMap[aggType] === 'Aggregations'
+                ? `buckets { key doc_count }`
+                : `stats { max min avg sum }`
+            }
+          }`;
+      })
+      .join('');
+  const query = `
+    {
+      ${ES_INDEX} {
+        aggregations { ${getAggregationQuery()} }
+      }
+    }`;
+  return fetchGraphqlQuery({
+    query,
+    ...fetchConfig,
+  }).then(data =>
+    Promise.resolve({
+      aggregations: Object.keys(data[ES_INDEX].aggregations).reduce(
+        (agg, key) => ({
+          ...agg,
+          [serializeToPath(key)]: data[ES_INDEX].aggregations[key],
+        }),
+        {},
+      ),
+    }),
+  );
+};
+
+export default class LiveAdvancedFacetView extends React.Component {
+  constructor(props) {
+    super(props);
+    const { sqon } = props;
+    this.state = {
+      mapping: {},
+      extended: {},
+      sqon: sqon || null,
+    };
+  }
+  componentDidMount() {
+    const { PROJECT_ID, ES_INDEX, API_HOST, ES_HOST } = this.props;
+    const fetchConfig = { PROJECT_ID, ES_INDEX, API_HOST, ES_HOST };
+    Promise.all([
+      fetchMapping(fetchConfig),
+      fetchExtendedMapping(fetchConfig),
+    ]).then(([{ mapping }, { extended }]) =>
+      fetchAggregationDataFromExtendedMapping({
+        extended,
+        ...fetchConfig,
+      }).then(({ aggregations }) =>
+        this.setState({ mapping, extended, aggregations }),
+      ),
+    );
+  }
+  componentWillReceiveProps({ sqon }) {
+    if (sqon) {
```

Comment:
It's possible an update to the sqon would make it `null` which this would block, so you should check for if the value has changed instead (you can use the lodash funtion `isEqual`)